js/web/lib/wasm/session-options.ts

```diff
@@ -75,6 +75,19 @@ const setExecutionProviders =
                   checkLastError(`Can't set a session config entry: 'deviceType' - ${webnnOptions.deviceType}.`);
                 }
               }
+              if (webnnOptions?.numThreads) {
+                let numThreads = webnnOptions.numThreads;
+                // Just ignore invalid webnnOptions.numThreads.
+                if (typeof numThreads != 'number' || !Number.isInteger(numThreads) || numThreads < 0) {
```

Comment:
Is `instanceof` better here, maybe avoiding the string compare? (I barely know any JS, but the deeper I dig into it, the messier and less consistent the rules appear 😅)

Reply:
'instaceof' should not be used for primitive types in JavaScript, e.g. 
```
1 instanceof Number; // false
typeof 1 == 'number'; // true
```

Reply:
> 1 instanceof Number; // false

🤦‍♂️ Oh Javascript...